lib/jekyll/admin/collections/handler.rb

```diff
@@ -0,0 +1,135 @@
+module Jekyll
+  module Admin
+    module Collections
+      class Handler < ApiHandler
+        # Create a new Handler.
+        # Inherits from ApiHandler to get common utility functions
+        #
+        # site - Jekyll::Site instance to run the commands at
+        #
+        # Returns nothing.
+        def initialize(site)
+          super(site)
+          @collections = @site.collections
+        end
+
+        # Get the values for collections INDEX route
+        #
+        # Returns an array of collection names.
+        def index_collections
+          @collections.keys
+        end
+
+        # Get the values for documents INDEX for a collection
+        #
+        # collection - name of the collection for which documents are to be extracted
+        #
+        # Returns an array of documents
+        #
+        # Documents have the following keys:
+        # collection_name - Name of the collection document is part of
+        # document_id - unique identifier for the document inside the collection
+        # meta - hash of metafields present in the document as frontmatter
+        def index_documents(collection)
+          documents = []
+          @collections[collection].docs.each do |doc|
+            meta = parse_frontmatter(doc.path)
+            doc_item = {
+              :collection_name => collection,
+              :document_id => doc.basename,
+              :meta => meta
+            }
+            documents << doc_item
+          end
+          documents
+        end
+
+        # SHOW a document inside a collection
+        #
+        # collection_name - name of the collection
+        # document_id - id of the document to be returned
+        #
+        # Returns a hash with details of the document
+        #
+        # Document has the following keys
+        # collection_name - Name of the collection document is part of
+        # document_id - unique identifier for the document inside the collection
+        # meta - hash of metafields present in the document as frontmatter
+        # body - contents of the document, except frontmatter (in html)
+        def show(collection_name, document_id)
+          collection = @collections[collection_name]
+          doc = find_document(collection, document_id)
+          if doc
+            meta = parse_frontmatter(doc.path)
+            document = {
+              :collection_name => collection_name,
+              :document_id => doc.basename,
```

Comment:
@parkr What is our final decision on these identifiers ? Should we make them only `id` ? Same as `page_id`.